src/safeposix/syscalls/net_calls.rs

```diff
@@ -1546,12 +1609,15 @@ impl Cage {
                                 );
                             }
 
-                            // simple if it's a domain socket
+                            //If the given socket is a Unix socket, lind handles
+                            //the connection, return with success
                             if sockhandle.domain == AF_UNIX {
                                 sockhandle.state = ConnState::LISTEN;
                                 return 0;
                             }
 
+                            //If the given socket is not assigned an address,
+                            //attempt to bind the socket to an address.
                             if sockhandle.localaddr.is_none() {
                                 let shd = sockhandle.domain as i32;
                                 let ibindret = self._implicit_bind(&mut *sockhandle, shd);
```

Comment:
Can you explain a bit about this function through a couple of comments for better understanding.